pkg/kubelet/volumemanager/volume_manager.go

```diff
@@ -232,11 +240,31 @@ type volumeManager struct {
 	// desiredStateOfWorldPopulator runs an asynchronous periodic loop to
 	// populate the desiredStateOfWorld using the kubelet PodManager.
 	desiredStateOfWorldPopulator populator.DesiredStateOfWorldPopulator
+
+	// the informer factory for CSIDriverLister
+	informerFactory  informers.SharedInformerFactory
+	csiDriversSynced kcache.InformerSynced
 }
 
 func (vm *volumeManager) Run(sourcesReady config.SourcesReady, stopCh <-chan struct{}) {
 	defer runtime.HandleCrash()
 
+	if utilfeature.DefaultFeatureGate.Enabled(features.CSIDriverRegistry) {
+		if vm.kubeClient != nil {
+			if vm.informerFactory != nil {
+				vm.informerFactory.WaitForCacheSync(wait.NeverStop)
```

Comment:
I think it's the wait for cache sync that's causing cluster up to fail. 

I'm trying to look into other examples inside kubelet of how they wait for the informer sync

Reply:
I could not find other examples inside kubelet that waits for the cache sync.

For now, can we move this to right before we want to access the informer?

Reply:
Sure.  I moved these to csi_plugin.go.